massa-execution/src/interface_impl.rs

```diff
@@ -230,16 +230,35 @@ impl Interface for InterfaceImpl {
     }
 
     fn generate_event(&self, data: String) -> Result<()> {
-        let context = context_guard!(self);
+        let mut context = context_guard!(self);
         let slot = context.slot;
         let block = context.opt_block_id;
         let call_stack = context.call_stack.clone();
+        let read_only = context.read_only;
+        if !read_only {
```

Comment:
always increment created_event_index AFTER generating it, even if we are in readonly